distributed/tests/test_spill.py

```diff
@@ -226,3 +227,50 @@ def __sizeof__(self):
             buf["a"] = a
 
     assert "Failed to pickle" in logs_bad_key.getvalue()
+
+    b = Bad(size=100)  # this is small enough to fit in memory/fast
+
+    buf["b"] = b
+    assert set(buf.fast) == {"b"}
+
+    c = 100 * "c"
+    with captured_logger(logging.getLogger("distributed.spill")) as logs_bad_key_mem:
+        # This will go to fast and try to kick b out,
+        # but keep b in fast since it's not pickable
+        buf["c"] = c
+
+    assert "Failed to pickle" in logs_bad_key_mem.getvalue()
+    assert set(buf.fast) == {"b", "c"}
+
+
+def test_spillbuffer_oserror(tmpdir):
+    buf = SpillBuffer(str(tmpdir), target=200, max_spill=800)
+
+    a, b, c = "a" * 200, "b" * 100, "c" * 200
+
+    # let's have something in fast and something in slow
+    buf["a"] = a
+    buf["b"] = b
+    assert set(buf.fast) == {"b"}
+    assert set(buf.slow) == {"a"}
+
+    # modify permissions of disk to be read only
+    subprocess.call(["chmod", "-R", "-w", str(tmpdir)])
+
+    with captured_logger(logging.getLogger("distributed.spill")) as logs_oserror:
+        buf["c"] = c
+
+    # PROBLEM: this is not triggering the OSError and even though c ends up in fast because
+    # "Spill disk reached capacity but it didn't", when
+    # I check buf.slow.total_weight the value corresponds as if c was there,
+    # buf.slow.weight_by_key = {'a': 329, 'c': 329} but set(buf.slow) = {"a"} and set(buf.fast) = {"b", "c"}
+
+    assert "Spill file to disk failed" in logs_oserror.getvalue()
```

Comment:
@crusaderky  Test is failing here (locally), in the assertion where I get the following error:
```python
>       assert "Spill file to disk failed" in logs_oserror_evict.getvalue()
E       AssertionError: assert 'Spill file to disk failed' in 'Spill file on disk reached capacity; keeping data in memory\n'
E        +  where 'Spill file on disk reached capacity; keeping data in memory\n' = <built-in method getvalue of _io.StringIO object at 0x11d95e9d0>()
E        +    where <built-in method getvalue of _io.StringIO object at 0x11d95e9d0> = <_io.StringIO object at 0x11d95e9d0>.getvalue

test_spill.py:276: AssertionError
```